frontend/src/scenes/onboarding/productAnalyticsSteps/DashboardTemplateVariables.tsx

```diff
@@ -0,0 +1,169 @@
+                    <p>
+                        Set the name that you'll use for a custom event (eg. a backend event) instead of selecting an
+                        event from your site.
+                    </p>
+                    <div className="flex gap-x-2 w-full">
+                        <LemonInput
+                            className="grow"
+                            onChange={(v) => {
+                                if (v) {
+                                    setCustomEventName(v)
+                                    setVariable(variable.name, {
+                                        events: [{ id: v, math: 'dau', type: 'events' }],
+                                    })
+                                } else {
+                                    setCustomEventName(null)
+                                    resetVariable(variable.id)
+                                }
+                            }}
+                            onBlur={() => {
+                                if (customEventName) {
+                                    setVariable(variable.name, {
+                                        events: [{ id: customEventName, math: 'dau', type: 'events' }],
+                                    })
+                                } else {
+                                    resetVariable(variable.id)
+                                    setShowCustomEventField(false)
+                                }
+                            }}
+                        />
+                        <div>
+                            <LemonButton
+                                icon={<IconTrash />}
+                                type="tertiary"
+                                size="small"
+                                onClick={() => {
+                                    resetVariable(variable.id)
+                                    setCustomEventName(null)
+                                    setShowCustomEventField(false)
+                                }}
+                            />
+                        </div>
+                    </div>
+                </div>
+            )}
+            {!hasSelectedSite ? (
+                <LemonBanner type="warning">Please select a site to continue. </LemonBanner>
+            ) : (
+                <div className="flex">
+                    {variable.touched ? (
+                        <LemonButton type="primary" status="alt" onClick={incrementActiveVariableIndex}>
+                            Continue
+                        </LemonButton>
+                    ) : (
+                        <div className="flex gap-x-2">
+                            <LemonButton
+                                type="primary"
+                                status="alt"
+                                onClick={() => {
+                                    setShowCustomEventField(false)
+                                    setVariable(variable.name, { events: [FALLBACK_EVENT] })
```

Comment:
in a follow-up PR this will actually get the action from the toolbar in the iframe